internal/querynode/tsafe_replica.go

```diff
@@ -23,38 +23,44 @@ import (
 
 // TSafeReplicaInterface is the interface wrapper of tSafeReplica
 type TSafeReplicaInterface interface {
-	getTSafe(vChannel Channel) Timestamp
-	setTSafe(vChannel Channel, id UniqueID, timestamp Timestamp)
+	getTSafe(vChannel Channel) (Timestamp, error)
+	setTSafe(vChannel Channel, id UniqueID, timestamp Timestamp) error
 	addTSafe(vChannel Channel)
-	removeTSafe(vChannel Channel)
-	registerTSafeWatcher(vChannel Channel, watcher *tSafeWatcher)
+	removeTSafe(vChannel Channel) error
+	registerTSafeWatcher(vChannel Channel, watcher *tSafeWatcher) error
+	removeRecord(vChannel Channel, id UniqueID) error
 }
 
 type tSafeReplica struct {
-	mu     sync.Mutex        // guards tSafes
-	tSafes map[string]tSafer // map[vChannel]tSafer
+	mu         sync.Mutex        // guards tSafes
+	tSafes     map[string]tSafer // map[vChannel]tSafer
```

Comment:
why tSafe use string as key but refCounter use vChannel as key

Reply:
if we make tSafes and refCounter in one map that would be better, like

type TSaferRef struct {
    a tSafer,
    ref int
}
map[Channel] Pair